core/lib/generators/solidus/install/install_generator.rb

```diff
@@ -61,23 +63,25 @@ def prepare_options
       @run_migrations = options[:migrate]
       @load_seed_data = options[:seed]
       @load_sample_data = options[:sample]
+      @selected_frontend = detect_frontend_to_install
+      @selected_authentication = detect_authentication_to_install
 
       # Silence verbose output (e.g. Rails migrations will rely on this environment variable)
       ENV['VERBOSE'] = 'false'
 
       # No reason to check for their presence if we're about to install them
       ENV['SOLIDUS_SKIP_MIGRATIONS_CHECK'] = 'true'
 
-      if options[:with_authentication] != nil
+      if options[:enforce_available_locales] != nil
         warn \
-          "DEPRECATION WARNING: using `solidus:install --with-authentication` is now deprecated. " \
-          "Please use --authentication instead (see --help for the full list of options)."
+          "DEPRECATION WARNING: using `solidus:install --enforce-available-locales` is now deprecated and has no effect. " \
+          "Since Rails 4.1 the default is `true` so we no longer need to explicitly set a value."
+      end
 
-        if options[:authentication].blank? && options[:with_authentication] == 'false'
-          # Don't use the default authentication if the user explicitly
-          # requested no authentication system.
-          options[:authentication] = 'none'
-        end
+      if options[:lib_name] != nil
+        warn \
+          "DEPRECATION WARNING: using `solidus:install --lib-name` is now deprecated and has no effect. " \
+          "The option is legacy and should be removed from scripts still using it."
```

Comment:
I'd like to understand... which kind of script could be using it?

Reply:
Scripts for generating the sandbox inside extensions are the most likely to have legacy options, also I think we had an example app to for the heroku button (not sure it's still maintained tho).

Over time would probably be safer to reject unknown options instead of silently accepting them, wdyt?

Reply:
Makes sense.

About rejecting unknown options, I think that should be the default.